test/datasets/test_dataset_metadata.py

```diff
@@ -1,523 +1,800 @@
-            assert instance.end_date == "March 27 2021"
-            assert instance.keywords == dataset_dict["dcat:keyword"]
-            assert all(isinstance(datafile, DataFile) for datafile in instance.files)
-            assert instance.themes == dataset_dict["dcat:theme"]
-            assert instance.publisher == dataset_dict["dct:publisher"]["foaf:name"]
-            assert instance.issued == "March 16 2021"
-            assert instance.rights == dataset_dict["dct:rights"]
-            assert instance.language == dataset_dict["dct:language"]
-            assert instance.standard == dataset_dict["dct:conformsTo"]["label"]
-            assert instance.update == dataset_dict["dct:accrualPeriodicity"]
+        self.assertEqual(metadata.rights, None)
+        self.assertEqual(metadata.update_frequency, None)
+        self.assertEqual(metadata.start_date, None)
+        self.assertEqual(metadata.end_date, None)
 
     @patch.object(DatasetMetadata, "output_metadata_extra_details")
     @patch.object(DatasetMetadata, "output_datafiles_table")
     @patch("dafni_cli.datasets.dataset_metadata.prose_print")
     @patch("dafni_cli.datasets.dataset_metadata.click")
-    class TestOutputMetadataDetails:
-        """Test class to test the output_metadata_details functionality"""
-
-        def test_data_outputted_as_expected(
-            self, mock_click, mock_prose, mock_table, mock_extra
-        ):
-            # SETUP
-            instance = dataset_meta_mock()
-
-            # CALL
-            instance.output_metadata_details()
-
-            # ASSERT
-            assert mock_click.echo.call_args_list == [
-                call(f"\nCreated: {instance.created}"),
-                call(f"Creator: {instance.creator}"),
-                call(f"Contact: {instance.contact}"),
+    def test_output_metadata_details(
+        self,
+        mock_click,
+        mock_prose,
+        mock_table,
+        mock_extra_details,
+    ):
+        """Tests output_metadata_details functions as expected"""
+
+        # SETUP
+        dataset_metadata: DatasetMetadata = parse_dataset_metadata(
+            TEST_DATASET_METADATA
+        )
+
+        # CALL
+        dataset_metadata.output_metadata_details()
+
+        # ASSERT
+        mock_click.echo.assert_has_calls(
+            [
+                call(f"\nCreated: {dataset_metadata.created}"),
+                call(f"Creator: {dataset_metadata.creators[0].name}"),
+                call(f"Contact: {dataset_metadata.contact}"),
                 call("Description:"),
-                call("Identifier: "),
-                call(f"Location: {instance.location}"),
-                call(f"Start date: {instance.start_date}"),
-                call(f"End date: {instance.end_date}"),
-                call(f"Key Words:\n {instance.keywords}"),
+                call("Identifiers:"),
+                call(f"Location: {dataset_metadata.location.label}"),
+                call(
```

Comment:
These assertions shouldn't really contain if statements. May want two tests that can test both situations